docs/interchange.rst

```diff
@@ -0,0 +1,478 @@
+    genealogical history.  Each mutation is associated with a particular
+    ``node`` (i.e., a particular ancestor), so that any sample which inherits
+    from that node will also inherit that mutation, unless another mutation
+    intervenes.  The type records::
+
+        site	node	derived_state
+        0	    14	    1
+
+    Here ``site`` is the index of the ``site`` at which the mutation occurred,
+    ``node`` records the ID of the ancestral node associated with the mutation,
+    and ``derived_state`` is the allele that any sample inheriting from that
+    node at this site will have if another mutation does not intervene.  The
+    ``node`` is not necessarily the ancestor in whom the mutation occurred, but
+    rather the ancestor at the bottom of the branch in the tree at that site on
+    which the mutation occurred.
+
+site
+    Rather than storing a position on the genome directly, a ``mutation``
+    stores the index of a ``site``, that describes that position.  This is to
+    allow efficient processing of multiple mutations at the same genomic
+    position.  A ``site`` records a position on the genome where a mutation has
+    occurred along with the ancestral state (i.e., the state at the root of the
+    tree at that position)::
+
+        id	position	ancestral_state
+        0	0.1	        0
+
+    As with nodes, the ``id`` is not stored directly, but is implied by its
+    index in the site table.
+
+
+To allow for efficent algorithms, it is required that
+
+    8. Sites are sorted by increasing position,
+    9. and mutations are sorted by site.
+
+Mutations at the same site may currently be in any order, but this may change.
+
+Migrations
+==========
+
+In simulations trees can be thought of as spread across space, and it is
+helpful for inferring demographic history to record this history.  This is
+stored using the following type.
+
+migration
+    Migrations are performed by individual ancestors, but most likely not by an
+    individual tracked as a ``node`` (as in a discrete-deme model they are
+    unlikely to be both a migrant and a most recent common ancestor).  So,
+    ``msprime`` records when a segment of ancestry has moved between
+    populations::
+
+        left    right   node    source  dest    time
+        0.0     0.3     3       0       1       2.1
+
+    This ``migration`` records that the ancestor who was alive 2.1 time units
+    in the past from which ``node`` 3 inherited the segment of genome between
+    0.0 and 0.3 migrated from population 0 to population 1.
+
+A valid ``migration``:
+
+1. Has ``time`` strictly between the time of its ``node`` and the time of any
+   ancestral node from which that node inherits on the segment ``[left,
+   right)``.
+2. Has the ``population`` of any such ancestor matching ``source``, if another
+   ``migration`` does not intervene.
+
+
+******************
+Text table formats
+******************
+
+.. todo Stuff about tables API.
```

Comment:
er, about the text formats? (that is `TreeSequence.dump_text` and `TreeSequence.load_text`)